js/ui/control/geolocate_control.js

```diff
@@ -104,13 +112,30 @@ class GeolocateControl extends Evented {
     }
 
     _onClickGeolocate() {
-
-        window.navigator.geolocation.getCurrentPosition(
-            this._onSuccess, this._onError, geoOptions);
-
-        // This timeout ensures that we still call finish() even if
-        // the user declines to share their location in Firefox
-        this._timeoutId = setTimeout(this._finish, 10000 /* 10sec */);
+        const positionOptions = util.extend(defaultGeoPositionOptions, this.options && this.options.positionOptions || {});
+
+        // toggle watching the device location
+        if (this.options.watchPosition) {
+            if (this._watching) {
```

Comment:
Good point. I've just changed it. The only caveat was that 0 seems to be a valid id (spec doesn't rule it out and Firefox Desktop seems to start at 0) so I need to explicitly check `!== undefined`.